README.md

```diff
@@ -177,8 +212,85 @@ public static void main(String[] args) throws Exception
+This is done through Gateway Events known as "dispatches" that are handled by the JDA WebSocket handlers.
+When Discord instructs JDA to delete entities they are simply removed from the JDA cache and lose their references.
+Once that happens nothing in JDA interacts or updates the instances of those entities and they become useless. Discord
+may instruct to delete these entities randomly for cache synchronization with the API.
+
+**It is not recommended to store _any_ of these entities for a longer period of time!**
+
+### Fake Entities
+
+Some entities in JDA are marked through an interface called `IFakeable`. These entities can exist outside
+of the JDA cache and are inaccessible through the common `get...ById(id)` methods.
+Fake entities are essentially instances that are not directly referenced by the JDA cached and are only
+temporarily created for a specific usage. It may be used for the author of a message that has left the guild
+when requesting the history of a `MessageChannel` or for `Emote` instances used in a `Message` that are not part
+of any of the guilds available to the bot.
+
+### Entity Updates
+
+When an entity is updated through its manager they will send a request to the Discord API which will update the state
+of the entity. The success of this request **does not** imply the entity has been updated yet. All entities are updated
+by the aforementioned **Gateway Events** which means you cannot rely on the cache being updated yet once the
+execution of a RestAction has completed. Some requests rely on the cache being updated to correctly update the entity.
+An example of this is updating roles of a member which overrides all roles of the member by sending a list of the
+new set of roles. This is done by first checking the current cache, the roles the member has right now, and appending
+or removing the requested roles. If the cache has not yet been updated by an event this will result in unexpected behavior.
+
+### Entity Deletion
+
+Discord may request that a client (the JDA session) invalidates its entire cache. When this happens JDA will
+remove all of its current entities and reconnect the session. This is signaled through the `ReconnectEvent`.
+When entities are removed from the JDA cache they lose access to the encapsulating entities. For instance
+a channel loses access to its guild. Once that happens they are unable to make any API requests through RestAction
+and instead throw an `IllegalStateException`. It is **highly recommended** to only keep references to entities
+by storing their **id** and using the respective `get...ById(id)` method when needed.
+
+#### Example
+
+```java
+public class UserLogger extends ListenerAdapter 
+{
+    private final long userId;
+    
+    public UserLogger(User user)
+    {
+        this.userId = user.getIdLong();
+    }
+    
+    @Override
+    public void onMessageReceived(MessageReceivedEvent event)
+    {
+        if (event.getAuthor().getIdLong() == userId)
```

Comment:
Store the author in a local variable to show that doing so is okay.

Reply:
We actually do this below in `onGuildJoin` but I guess we can do it here as well.